### ISSUE TITLE
Add created_at/updated_at timestamp

### ISSUE THREAD

**Issue by joneslee85** [lotus/model#134]
What: Having a conventional / implicit timestamp plugin
How:
  * Implement a special attribute helper for collection declaration, eg:
  ```ruby
  collection :users do
    entity     User
    repository UserRepository

    attribute :id,   Integer
    attribute :name, String
    timestamps
  end
  ```

* Enable [Sequel::Plugins::Timestamps](http://sequel.jeremyevans.net/rdoc-plugins/classes/Sequel/Plugins/Timestamps.html) If the DB table has column `created_at` and `updated_at`

Pending for discussion

**Comment by jodosha** [comment]
@joneslee85 Doing this via coercer is a bad idea :( It can work when the value is `nil` (`created_at` on a non-persisted entity). But when it comes to update an already existing timestamp (`updated_at`), it won't replace it with a default value.

**Comment by AlfonsoUceda** [comment]
I think the default option should be in the migrations and not in the coercer.

And only care here about include this timestamps

**Comment by angeloashmore** [comment]
@AlfonsoUceda Right, migration-less/schema-less databases (e.g. a JSON store) don't have default values or types. Although there are ways to coerce data to certain data types natively within some of these databases, not every database has that functionality. It would have to take place at the application level.

Is there a reason `#create` only updates the id versus the whole entity reference? My guess is because the database interface only returns that value upon creation?

```ruby
# sql_adapter.rb
def create(collection, entity)
  entity.id = command(
                query(collection)
              ).create(entity)
  entity
end
```

**Comment by jodosha** [comment]
@AlfonsoUceda @angeloashmore This isn't only a matter of schema(less) databases. If a column isn't mapped it's entirely ignored by the framework.

@angeloashmore RE entity: that is my fault. I'm sorry. I already discussed with @joneslee85 about this problem. You have opened a ticket for this, let's discuss over there.

**Comment by angeloashmore** [comment]
It appears the Repository is the best place for this to happen. Maintaining timestamps should occur at `#persist`.

I wrote this quick module and it seems to work. I haven't written any tests for it yet, however.

https://gist.github.com/angeloashmore/05ef7f035f694cfa56d7

## Required setup
1. The database must accept `created_at` and `updated_at`.
2. The entity must respond to `#created_at=` and `#updated_at=`.
3. The repository must set `created_at` and `updated_at` before persisting/creating/updating.
4. The mapper must include `created_at` and `updated_at`.
5. The repository must `include Lotus::Repository::Timestamps`.

```ruby
# order_repository.rb
class OrderRepository
  include Lotus::Repository
  include Lotus::Repository::Timestamps
end
```

```ruby
# mapping.rb
mapping do
  collection :orders do
    entity Order

    attribute :id, String
    attribute :billing_info, Hash
    attribute :created_at, Time
    attribute :updated_at, Time
  end
end
```

```ruby
# order.rb
class Order
  include Lotus::Entity
  attributes :billing_info, :created_at, :updated_at
end
```

## Issues with this approach
There is a lot of repetition of `created_at` and `updated_at`. Because of the way Repositories, Mappers, and Entities are separated, it seems it is necessary to declare the attributes in all places.

What do you think of having helper methods to automatically include this? You will still need to declare them everywhere, but as @joneslee85 suggested, it could turn this:
```ruby
collection :orders do
  # ...
  attribute :created_at, Time
  attribute :updated_at, Time
end
```
…to this:
```ruby
collection :orders do
  # ...
  timestamps
end
```
…and this:
```ruby
class Order
  include Lotus::Entity
  attributes :billing_info, :created_at, :updated_at
end
```
…to this:
```ruby
class Order
  include Lotus::Entity
  attributes :billing_info, timestamps
end
```

**Comment by jodosha** [comment]
@angeloashmore Thanks for the detailed investigation. I apologize for the late reply.

I do agree that there is a lot of repetition. I suggest to make it work and then to make it DRY. Would you love to craft a PR based on that gist? Thank you very much!

**Comment by dsnipe** [comment]
Is anybody works on this feature? 
I can do task based on a proposed solution and then continue implementing this feature, if not.

**Comment by AlfonsoUceda** [comment]
@dsnipe there isn't someone working on this, please feel free ;)

**Comment by dsnipe** [comment]
@AlfonsoUceda cool :) 
 so, I done with provided gist in PR 
My plan is:
* Implement *timestamps* helpers for mapper and entity
* Equivalency for entities. We can use https://github.com/dkubb/equalizer or create our own solution to avoid dependency.

**Comment by AlfonsoUceda** [comment]
avoid new dependencies please, create a custom solution for lotus ;)

thanks!

**Comment by dsnipe** [comment]
ok, I think I'll place it in lotus/utils, it's better place for it I suppose

**Comment by dsnipe** [comment]
After creating draft for timestamp implementation, I realised, that may be better place for defining timestamps will be in configuration.
### Global defining for all entities
```ruby
Lotus::Model.configure do
  timestamps :all
end
```
### Global defining with exceptions
```ruby
Lotus::Model.configure do
  timestamps :all, exclude: ['Admin', 'Moderator']
end
```
### Defining timestamps only for specific entities
```ruby
Lotus::Model.configure do
  timestamps ['User', 'Article', 'Blog']   # not sure about view of this one
end
```
It gives ability to manage timestamps from one place and generate migrations already with needed columns. 
What are think about it?

**Comment by joneslee85** [comment]
IMHO I think your suggestion is over killing for I don't find much gains in doing so. I'd rather let users self-define in their entity for sake of consistency and simplicity. Just my 5cent

**Comment by dsnipe** [comment]
@joneslee85 actually I used timestamps almost on all entities usually. That why I suggest using it in global scope. 
but maybe you're right and simplest way is a better way :) I'll finish #169 
btw,  I see some duplication of functionality in #167 and this one – there are any reasons that both solutions are necessary?

**Comment by joneslee85** [comment]
@dsnipe much thanks, regarding #167, it is overlapping, however I'd like to ask you to wait for TeamLeo who is now working on it. :+1:

**Comment by jodosha** [comment]
Implemented by #173 #176 #182
